### PR TITLE
enabled ineffassign in golangci

### PR DESCRIPTION
**Pull request by skoef**
In line with #663, #664 and #665, I enabled the ineffassign linter and fixed some small issues that came from it.

### CODE REVIEW

File: mysql/mysql_test.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ func (t *mysqlTestSuite) TestMysqlGTIDInterval(c *check.C) {
 
 	i, err = parseInterval("1-2")
```
Comment (atercattus) [inline mysql/mysql_test.go:41]: I see no difference between this test and the first. Maybe we should delete this test?

File: mysql/rowdata.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ func (p RowData) ParseText(f []*Field, dst []FieldValue) ([]FieldValue, error) {
 	var v []byte
 	var isNull bool
 	var pos int = 0
-	var n int = 0
+	var n int
```
Comment (atercattus) [inline mysql/rowdata.go:30]: Maybe remove zero value for `pos` too?

File: client/resp.go [hunk 3]
```diff
@@ -53,14 +53,13 @@ func (c *Conn) handleOKPacket(data []byte) (*Result, error) {
...
 		r.Warnings = binary.LittleEndian.Uint16(data[pos:])
-		pos += 2
 	} else if c.capability&CLIENT_TRANSACTIONS > 0 {
 		r.Status = binary.LittleEndian.Uint16(data[pos:])
 		c.status = r.Status
-		pos += 2
 	}
 
 	//new ok package will check CLIENT_SESSION_TRACK too, but I don't support it now.
+	// pos += 2
```
Comment (atercattus) [inline client/resp.go:62]: It seems to me that it is still worth leaving the change of position next to the reading, and not taking it outside. Any future refactoring or code change can either make the code harder to understand or break the code.

File: mysql/mysql_test.go (hunk 1 above)
Comment (skoef) [reply mysql/mysql_test.go]: there is none, I didn't notice the earlier test. I deleted it.

File: mysql/rowdata.go (hunk 2 above)
Comment (skoef) [reply mysql/rowdata.go:30]: sure

File: client/resp.go (hunk 3 above)
Comment (skoef) [reply client/resp.go]: I've placed them back and commented them, I think that is the most readable way

**Review by atercattus — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Comment by skoef** [comment]
Thanks for the approval @lance6716 and @atercattus ! But I can't merge in this repo so if wou want to do this, thanks!

**Pull request merged**